### PR TITLE
Add FXIOS-8490 [v124] Add `AutofillTelemetry` to list of files pulled from mozilla-central

### PR DESCRIPTION
**Pull request by issammani**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-8490)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/18854)

## :bulb: Description
This PR:
- Adds `AutofillTelemetry` file to list of files pulled from m-c
- Adds `addressFormTelemetryMessageHandler` as it's needed by  https://github.com/mozilla-mobile/firefox-ios/pull/18842
- Blocks  https://github.com/mozilla-mobile/firefox-ios/pull/18842

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] ~Wrote unit tests and/or ensured the tests suite is passing~
- [ ] ~When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)~
- [ ] ~If needed, I updated documentation / comments for complex code and public methods~
- [ ] ~If needed, added a backport comment (example `@Mergifyio backport release/v120`)~

### CODE REVIEW

File: firefox-ios/Client/Frontend/TabContentsScripts/FormAutofillHelper/FormAutofillHelper.swift
Comment (issammani) [inline firefox-ios/Client/Frontend/TabContentsScripts/FormAutofillHelper/FormAutofillHelper.swift:91]: @nbhasin2 this is needed because https://github.com/mozilla-mobile/firefox-ios/pull/18842 assumes this handler exists. @jnrahme will work on the details of the implementation later.

Comment (nbhasin2) [reply firefox-ios/Client/Frontend/TabContentsScripts/FormAutofillHelper/FormAutofillHelper.swift:91]: nit: the category could be `address` or you can add a new one for `formAutofill`

**Review by nbhasin2 — APPROVED** [review]

File: firefox-ios/Client/Frontend/TabContentsScripts/FormAutofillHelper/FormAutofillHelper.swift
Comment (nbhasin2) [reply firefox-ios/Client/Frontend/TabContentsScripts/FormAutofillHelper/FormAutofillHelper.swift:91]: autofill can be a much bigger category and since you are having side discussion we can merge this one as is @lmarceau @issammani as I don't want to block this PR if something will get changed or added in future

Comment (lmarceau) [reply firefox-ios/Client/Frontend/TabContentsScripts/FormAutofillHelper/FormAutofillHelper.swift:91]: To unblock this PR, we can use the `address` logger category for now

Comment (issammani) [reply firefox-ios/Client/Frontend/TabContentsScripts/FormAutofillHelper/FormAutofillHelper.swift:91]: Ohhh I had an outdated tab open. Making the change now 😄

Comment (issammani) [reply firefox-ios/Client/Frontend/TabContentsScripts/FormAutofillHelper/FormAutofillHelper.swift]: Done !

**Review by lmarceau — APPROVED** [review]

**Comment by nbhasin2** [comment]
Br Failed - https://app.bitrise.io/build/d9356636-3e2e-4680-b709-ab23daa59126

@issammani

**Comment by issammani** [comment]
> Br Failed - https://app.bitrise.io/build/d9356636-3e2e-4680-b709-ab23daa59126
> 
> @issammani

Should be fixed in [090804c](https://github.com/mozilla-mobile/firefox-ios/pull/18857/commits/090804cae52b1df52e07d55f7b025b1778ac7a8e)

**Comment by nbhasin2** [comment]
@Mergifyio backport release/v124

**Comment by nbhasin2** [comment]
@Mergifyio backport release/v124

**Pull request merged**
